### PR TITLE
[W5.11][T13-1] Adrian Tan

### PR DESCRIPTION
**Pull request by QzSG**
[[LO-2KLoC]](https://github.com/nusCS2113-AY1819S1/addressbook-level2/blob/master/docs/LearningOutcomes.adoc#work-in-a-2kloc-code-base-lo-2kloc) Add a feature to AddressBook
- Adds `sort` command to Addressbook-2 along with relevant unit tests to fulfill `W5.11` 🏆

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (ewaldhew) [inline src/seedu/addressbook/parser/Parser.java]: `import` statements should be explicit. Configure your IntelliJ to not collapse imports like this.
`Preferences` > `Editor` > `Code Style` > `Java` > `Imports tab set Class count to use import with '*'`

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Test
+    public void sortCommand_sortsAddressBookCorrectly() {
+        SortCommand sortCommand = new SortCommand();
+        SortCommand redundantCommand = new SortCommand();
+        sortCommand.setData(unsortedAddressBook, listWithUnsortedPersons);
+
+        // sortedAddressBook is already sorted when created.
+        SortCommand expectedCommand = new SortCommand();
```
Comment (ewaldhew) [inline test/java/seedu/addressbook/commands/SortCommandTest.java]: This is never used..

**Review by ewaldhew — COMMENTED** [review]
Looks fine to me, but do remember to configure coding style in IntelliJ. Close this PR after reading.

File: test/java/seedu/addressbook/commands/SortCommandTest.java (hunk 2 above)
Comment (QzSG) [reply test/java/seedu/addressbook/commands/SortCommandTest.java]: Looks like I left artifacts when renaming stuff, thanks for the heads up...

**Comment by QzSG** [comment]
Closing PR

**Comment by luhan02** [comment]
Great enhancement.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public class SortCommand extends ListCommand {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book as a list with index numbers sorted by name.\n"
+            + "Example: " + COMMAND_WORD;
+
+
+
```
Comment (ewaldhew) [inline src/seedu/addressbook/commands/SortCommand.java:23]: Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -72,6 +72,12 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Sorting all persons : `sort`
+
+Sorts all persons and persists to storage.
+Also shows a list of all persons, along with their non-private details, in the address book after sorting. +
+Format: `sort`
+
```
Comment (ewaldhew) [inline docs/UserGuide.adoc:80]: Good job updating the user guide

**Review by ewaldhew — COMMENTED** [review]
Great work, and with updating the JUnit tests as well. 👍 Close PR after reading

**Comment by QzSG** [comment]
Closing PR

**Pull request closed without merging**